lib/hashes.md

```diff
@@ -5,40 +5,42 @@ instructor_notes: Feel free to re-organize the headings (or add/remove headings)
 
 # What is a Hash?
 
-Replace me with your response...
+A hash is the python form of a dictionary.  They are similar to arrays, but whereas an array uses an integer to index and organize the objects located within it, whereas a hash is more variable.  It can use any value to help organize
+the list.  Hashes can help group different items/objects in the list into associations.  Like what I know from python, the key is the sorting, association-making variables that help group, label, and identify the values assigned to them.
 
 # What are some examples of information that would be Hashes as opposed to some other data type?
 
-Replace me with your response...
+Since hashes do not store the items in a numbered list, so if you need the data stored in a specific order, so that it can be called by referencing a specific number in that sequence, then hashes can be useful.  They can group and organize data/objects,
+and make it so that items can be called without havinig to memorize their specific number in the ordered list or array.  Arrays store named data, so the use of the name can do a lot of things with the data associated with that key.  A type of data that can 
+benefit from the use of hashes is counting data, as it can store amounts of a certain type of item (as opposed to arrays).
 
 # How are Hashes and Arrays similar? How are they different?
 
-Replace me with your response...
-
+Some of this response could be found in the previous answer above, but I'll try to repeat as much here.  Both arrays and hashes are collections of information, stored in a form of repository.  The differences come mainly from how that data is stored, and how that data can be used or called.
+Arrays store the data/objects in a specific order, and particular items can be called by utilizing the number of the item in that list (beginning with zero).  Hashes have keys, which are unique names that can be used to identify values.  The keys can be used to help do a variery of things, like 
+organize a count of like items, or categorize items into desired groups, etc.  Hashes have much more variety in terms of what they can offer.
 # How do you retrieve a particular value from a Hash?
 
-Replace me with your response...
-
+A particular value from a hash is retrieved by using the key for that value.  For instance, if this is our hash:  grades = { "Mike Ducey" => A, "Pat Ducey" => B+ }, you reference the hash like an array, but instead of specifying the number of the item, you use the key.  So, in this case, if I write grades["Mike Ducey"], I shall get => A as a response.
```

Comment:
No, that would not work, as those values need to be in quotation marks.  Those objects, without quotation marks, would cause errors, especially because it reads the + as a method and tries to add!  I think if I don't use integers or floats, I'll have to be sure to use quotation marks.